pkg/prometheus/operator.go

```diff
@@ -1295,31 +1341,35 @@ func PrometheusStatus(ctx context.Context, kclient kubernetes.Interface, p *moni
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "retrieving pods of failed")
 	}
-	sset, err := kclient.AppsV1().StatefulSets(p.Namespace).Get(ctx, statefulSetNameFromPrometheusName(p.Name), metav1.GetOptions{})
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "retrieving stateful set failed")
-	}
-
-	res.Replicas = int32(len(pods.Items))
 
 	var oldPods []v1.Pod
-	for _, pod := range pods.Items {
-		ready, err := k8sutil.PodRunningAndReady(pod)
+	expected := expectedStatefulSetShardNames(p)
```

Comment:
Do we need this check again?

Reply:
I wanted status to be entirely detached from the "actionable code", kind of like a black-box or unit test, testing for a specific pre-calculated result.